### PR TITLE
attributes: add upstream.cx_pool_ready_duration to get the duration of the upstream connection pool ready duration

### PR DESCRIPTION
**Pull request by agrawroh**
**Commit Message**: add `upstream.cx_pool_ready_duration` to get the duration of the upstream connection pool ready duration
**Additional Description**: Added a new attribute to extract `upstream.cx_pool_ready_duration` from the `StreamInfo()`. This could be used in CEL Matcher for access log filtering.
**Risk Level**: Low
**Testing:** Added Unit Tests
**Docs Changes:** Added
**Release Notes:** Added

### CODE REVIEW

**Comment by agrawroh** [comment]
cc @wbpcode

File: source/extensions/filters/common/expr/context.cc [hunk 1]
```diff
@@ -367,6 +367,22 @@ const UpstreamLookupValues& UpstreamLookupValues::get() {
...
                  wrapper.info_.upstreamInfo().value().get().upstreamTransportFailureReason());
            }},
+          {UpstreamConnectionPoolReadyDuration,
+           [](const UpstreamWrapper& wrapper) -> absl::optional<CelValue> {
+             if (!wrapper.info_.upstreamInfo().has_value()) {
+               return {};
+             }
+             if (auto connection_pool_callback_latency = wrapper.info_.upstreamInfo()
+                                                             .value()
+                                                             .get()
```
Comment (wbpcode) [inline source/extensions/filters/common/expr/context.cc:377]: replace this with `->` directly.

File: source/extensions/filters/common/expr/context.cc [hunk 2]
```diff
@@ -367,6 +367,22 @@ const UpstreamLookupValues& UpstreamLookupValues::get() {
              return CelValue::CreateStringView(
                  wrapper.info_.upstreamInfo().value().get().upstreamTransportFailureReason());
            }},
+          {UpstreamConnectionPoolReadyDuration,
+           [](const UpstreamWrapper& wrapper) -> absl::optional<CelValue> {
+             if (!wrapper.info_.upstreamInfo().has_value()) {
+               return {};
+             }
+             if (auto connection_pool_callback_latency = wrapper.info_.upstreamInfo()
```
Comment (wbpcode) [inline source/extensions/filters/common/expr/context.cc:375]: nit: const and explicit type is prefered is the type name is not too long.

**Review by wbpcode — COMMENTED** [review]
LGTM overall with some minor comments

File: source/extensions/filters/common/expr/context.cc (hunk 1 above)
Comment (agrawroh) [reply source/extensions/filters/common/expr/context.cc:377]: CHanged, Thanks!

**Review by wbpcode — APPROVED** [review]

**Pull request merged**